### PR TITLE
Added PWA

### PR DESCRIPTION
**Pull request by TubaApollo**
Would be cool if you, tasky, can check if this works out? I tested it and it seems to work, but idk if its fine like this. I am kinda worried of issues with the reload dialogue, not really sure if it works completely fine, and maybe style the reload dialogue in ReloadPrompt.vue so that it looks better.

### CODE REVIEW

File: .vitepress/config.mts [hunk 1]
```diff
@@ -145,4 +146,43 @@ export default defineConfig({
       { icon: "discord", link: "https://discord.gg/WYchhG8z8T" },
     ],
   },
-});
+
+  pwa: {
+    mode: 'development',
+    base: '/',
+    scope: '/',
+    includeAssets: ['favicon.svg'],
```
Comment (TubaApollo) [inline .vitepress/config.mts:154]: also not sure about this since their obviously is no favicon.svg and idk where its located

Comment (taskylizard) [reply .vitepress/config.mts:154]: Try this: https://github.com/anotherduckling/vite/blob/2d3591fc8dd47ef0f491e987e0a3201b821f0fe8/.vitepress/config.mts#L26

**Comment by taskylizard** [comment]
Did you actually test these changes? `@vite-pwa/vitepress` is nowhere added to package.json.

**Comment by TubaApollo** [comment]
I did but haven't pushed the package file with it.. 🥲 it builds fine and works on the test server since it has the package installed.

**Comment by TubaApollo** [comment]
I will add the package file later

**Comment by taskylizard** [comment]
I have made the changes locally, lots of missing bits but will just finish it off.

**Comment by TubaApollo** [comment]
Thanks.

**Pull request merged**